src/faultlog/util.cpp

```diff
@@ -182,4 +184,58 @@ json parseCallout(const std::string callout)
     return sectionJson;
 }
 
+bool isECOModeEnabled(struct pdbg_target* coreTgt)
+{
+    ATTR_ECO_MODE_Type ecoMode;
+    if (DT_GET_PROP(ATTR_ECO_MODE, coreTgt, ecoMode))
+    {
+        return false;
+    }
+
+    if (ecoMode == ENUM_ATTR_ECO_MODE_ENABLED)
+    {
+        return true;
+    }
+    return false;
+}
+
+bool isECOcore(struct pdbg_target* target)
+{
+    const char* tgtClass = pdbg_target_class_name(target);
+    if (!tgtClass)
+    {
+        return false;
```

Comment:
Could a error_log here help in debugging?

Reply:
Agreed with @deepakala-k